src/ui/components/filters/filterboxcomponent.js

```diff
@@ -238,11 +248,15 @@ export default class FilterBoxComponent extends Component {
    * Handle changes to child filter components
    * @param {number} index The index of the changed filter
    * @param {FilterNode} filterNode The new filter node
+   * @param {boolean} saveFilterNodes Whether to save filternodes to storage
+   * @param {boolean} searchOnChange Whether to conduct a search
    */
-  onFilterNodeChange (index, filterNode) {
+  onFilterNodeChange (index, filterNode, saveFilterNodes, searchOnChange) {
```

Comment:
It's possible that someone could have saveFilterNodes as false and searchOnChange as true. Is that a problematic state? They'd issue a query without actually saving the updated nodes in storage first. It seems like they'd always want to saveFilterNodes if searchOnChange is true.

Reply:
sure I think that makes intuitive sense even if it's slightly more complicated